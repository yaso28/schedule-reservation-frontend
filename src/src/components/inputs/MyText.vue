<template>
  <CInput
    :id="id"
    horizontal
    :type="type"
    :label="label"
    :is-valid="validate"
    :invalid-feedback="invalidMessage"
    :value="value"
    @input="setValue"
  />
</template>

<script>
export default {
  name: 'MyText',
  mixins: [require('./input-mixin').default],
  props: {
    type: {
      type: String,
      default: 'text'
    }
  }
}
</script>
