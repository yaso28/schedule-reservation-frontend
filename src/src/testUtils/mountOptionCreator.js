import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
