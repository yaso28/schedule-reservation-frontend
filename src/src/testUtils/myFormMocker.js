import MyForm from '@/models/MyForm'
jest.mock('@/models/MyForm')

class MyFormMocker {
  createMockSingleField (field, initValue) {
    const input = {}
    input[field] = initValue
    return this.createMock(input)
  }

  createMock (input) {
    const mockForm = new MyForm(input)
    mockForm.values = input
    mockForm.errors = {}
    return mockForm
  }

  async setValue (wrapper, form, field, value) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        values: Object.assign({}, form.values, { [field]: value })
      })
    })
  }

  setValueNonProps (form, field, value) {
    form.values[field] = value
  }

  async setError (wrapper, form, field, errorMessage) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        errors: Object.assign({}, form.errors, { [field]: errorMessage })
      })
    })
  }
}

export default new MyFormMocker()
