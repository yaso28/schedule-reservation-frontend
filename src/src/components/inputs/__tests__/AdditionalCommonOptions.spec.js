import { mount } from '@vue/test-utils'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import MyText from '../MyText'
import myFormMocker from '@/testUtils/myFormMocker'

jest.mock('@/models/MyForm')

function getInputWrapper (wrapper) {
  return wrapper.find('input')
}

describe('AdditionalCommonOptions', () => {
  let label
  let form
  let field
  let mountOption
  let wrapper
  beforeEach(() => {
    label = 'サンプル'
    mountOption = mountOptionCreator.create()
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('IdPrefix', () => {
    beforeEach(() => {
      field = 'sample'
      form = myFormMocker.createMockSingleField(field, '')
      mountOption.propsData = { label, form, field, idPrefix: 'prefix' }
      wrapper = mount(MyText, mountOption)
    })

    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Nested', () => {
    beforeEach(() => {
      field = 'sample.1.sample_field'
      form = myFormMocker.createMockSingleField('sample', [
        { sample_field: '' },
        { sample_field: '' }
      ])
      mountOption.propsData = { label, form, field }
      wrapper = mount(MyText, mountOption)
    })

    describe('通常', () => {
      test('表示', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('バリデーションエラー', () => {
      beforeEach(async () => {
        await myFormMocker.setError(wrapper, form, field, 'エラーです')
      })

      test('表示', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('入力', () => {
      const newValue = 'あいう'
      beforeEach(async () => {
        const inputWrapper = getInputWrapper(wrapper)
        await inputWrapper.setValue(newValue)
      })

      test('MyForm.setValue()を呼出', () => {
        expect(form.setValue).toHaveBeenCalledWith(field, newValue)
      })
    })
  })
})
