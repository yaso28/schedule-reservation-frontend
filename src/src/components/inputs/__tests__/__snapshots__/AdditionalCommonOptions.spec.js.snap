// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalCommonOptions IdPrefix 表示 1`] = `
<div
  class="form-group form-row"
  role="group"
>
  <label
    class="col-form-label col-sm-3"
    for="prefix-sample"
  >
     サンプル 
  </label>
  <div
    class="col-sm-9"
  >
    <!---->
    <input
      class="form-control"
      id="prefix-sample"
      type="text"
    />
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;

exports[`AdditionalCommonOptions Nested バリデーションエラー 表示 1`] = `
<div
  class="form-group form-row"
  role="group"
>
  <label
    class="col-form-label col-sm-3"
    for="sample.1.sample_field"
  >
     サンプル 
  </label>
  <div
    class="col-sm-9"
  >
    <!---->
    <input
      class="form-control is-invalid"
      id="sample.1.sample_field"
      type="text"
    />
    <!---->
    <!---->
    <div
      class="invalid-feedback"
    >
      エラーです
    </div>
    <!---->
  </div>
</div>
`;

exports[`AdditionalCommonOptions Nested 通常 表示 1`] = `
<div
  class="form-group form-row"
  role="group"
>
  <label
    class="col-form-label col-sm-3"
    for="sample.1.sample_field"
  >
     サンプル 
  </label>
  <div
    class="col-sm-9"
  >
    <!---->
    <input
      class="form-control"
      id="sample.1.sample_field"
      type="text"
    />
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
</div>
`;
